ide-support/revsblibrary.livecodescript

```diff
@@ -986,12 +989,733 @@ function revSBLibraryList
+      -- to check if the platform is supported
+      repeat for each line tFolder in tFolders
+         local tContents
+         put __revSBEnumerateFolder(tFolder, tFolder & slash) into tContents
+         
+         local tRawList
+         switch pPlatform
+            case "ios"
+            case "android"
+               put revSBFilterExternalsForMobile(tContents, pPlatform) into tRawList
+               break
+            case "windows"
+            case "macosx"
+            case "linux"
+               put revSBFilterExternalsForDesktop(tContents, pPlatform) into tRawList
+               break
+            default
+               put empty into tRawList
+               break
+         end switch
+         
+         split tRawList by return
+         set the itemdelimiter to "."
+         set the linedelimiter to slash
+         repeat for each element tExtPath in tRawList
+            put tExtPath into sExternalsLocationA[pPlatform][item 1 of line -1 of tExtPath]
+         end repeat
+      end repeat
+   end if
+   return the keys of sExternalsLocationA[pPlatform]
+end revSBAdditionalExternalsList
+
+-- Instead of resolving the path to an included external each time
+-- fetch the stored path from a script local array
+function revSBGetExternalPath pPlatform, pName
+   if sExternalsLocationA is empty then
+      get revSBExternalsList(pPlatform)
+   end if
+   
+   return sExternalsLocationA[pPlatform][pName]
+end revSBGetExternalPath
+
+-- Ensure the name used for platforms is consistent
+constant kMacPlatform = "MacOSX"
+constant kWindowsPlatform = "Windows"
+constant kLinuxPlatform = "Linux"
+constant kiOSPlatform = "iOS"
+constant kAndroidPlatform = "Android"
+constant kEmscriptenPlatform = "Emscripten"
+
+function revSBAllPlatforms
+   return kMacPlatform & comma & \
+         kWindowsPlatform & comma & \
+         kLinuxPlatform & comma & \
+         kiOSPlatform & comma & \
+         kAndroidPlatform & comma & \
+         kEmscriptenPlatform
+end revSBAllPlatforms
+
+function revSBAllDesktopPlatforms
+   return kMacPlatform & comma & \
+         kWindowsPlatform & comma & \
+         kLinuxPlatform
+end revSBAllDesktopPlatforms
+
+function revSBAllMobilePlatforms
+   return kiOSPlatform & comma & \
+         kAndroidPlatform
+end revSBAllMobilePlatforms
+
+private command addToList pElement, @xArray
+   put pElement into xArray[the number of elements in xArray + 1]
+end addToList
+
+private function revSBAvailableWidgets
+   return revSBAvailableExtensions("widget")
+end revSBAvailableWidgets
+
+private function revSBAvailableLibraries
+   return revSBAvailableExtensions("library")
+end revSBAvailableLibraries
+
+-- LCB Extensions
+private function revSBAvailableExtensions pType
+   local tExtensions, tInclusions
+   put revIDEExtensions(pType, "installed") into tExtensions
+   repeat for each key tID in tExtensions
+      local tExtension
+      -- LCB extensions are available on all platforms
```

Comment:
I think we need to add a note to bug 16535 ("Include extensions' code modules in standalones") to update this, because once extensions can contain native code this will no longer be true.